### PR TITLE
segmentation tutorial

### PR DESCRIPTION
**Pull request by nirhutnik**

### CODE REVIEW

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 1]
```diff
@@ -27,151 +27,108 @@
...
+#   or other frameworks, please visit the :ref:`creating VisionData guide <vision_data__creating_vision_data>`.
 
-# Importing the required packages
-import matplotlib.pyplot as plt
-import numpy as np
-import torch
-import torchvision.transforms.functional as F
-
-
-# %%
```
Comment (noamzbr) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:38]: Keep this `# %%`, otherwise the following text is rendered as a comment rather than rst text

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 2]
```diff
@@ -27,151 +27,108 @@
 # %%
 # Defining the data and model
 # ===========================
+# .. note::
+#   In this tutorial, we use the pytorch to create the dataset and model. To see how this can be done using tensorflow
+#   or other frameworks, please visit the :ref:`creating VisionData guide <vision_data__creating_vision_data>`.
```
Comment (noamzbr) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:32]: I think this is not the correct way to link. Try
```suggestion
#   or other frameworks, please visit the :ref:`creating VisionData guide </user-guide/vision/VisionData>`.
```

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 3]
```diff
@@ -27,151 +27,108 @@
...
-# The model in this tutorial is used to detect different objects in images (labels based on the Pascal VOC dataset).
+# The model in this tutorial is used to detect different object segments in images (labels based on the Pascal VOC dataset).
 # The model is trained to identify 20 different objects (person, bicycle etc.) and background.
 # The dataset itself is the COCO128 dataset with semantic segmentation labels, mapped to the Pascal VOC labels
-# (Originally, the COCO dataset includes more labels, but those has been filtered out)
-# The dataset can be loaded as a pytorch DataLoader object from deepchecks.vision.datasets.segmentation, as is done in
-# this tutorial, but can also be loaded as a SegmentationData object
+# (Originally, the COCO dataset includes more labels, but those have been filtered out).
+# The dataset can be loaded as a pytorch Dataset object from deepchecks.vision.datasets.segmentation, as is done in
+# this tutorial, but can also be loaded as a Vision object using the "load_dataset" function from that directory,
```
Comment (noamzbr) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:41]:
```suggestion
# this tutorial, but can also be loaded as a VisionData object using the "load_dataset" function from that directory,
```

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 4]
```diff
@@ -27,151 +27,108 @@
...
 # ~~~~~~~~~
-# The model in this tutorial is used to detect different objects in images (labels based on the Pascal VOC dataset).
+# The model in this tutorial is used to detect different object segments in images (labels based on the Pascal VOC dataset).
 # The model is trained to identify 20 different objects (person, bicycle etc.) and background.
 # The dataset itself is the COCO128 dataset with semantic segmentation labels, mapped to the Pascal VOC labels
-# (Originally, the COCO dataset includes more labels, but those has been filtered out)
-# The dataset can be loaded as a pytorch DataLoader object from deepchecks.vision.datasets.segmentation, as is done in
-# this tutorial, but can also be loaded as a SegmentationData object
+# (Originally, the COCO dataset includes more labels, but those have been filtered out).
+# The dataset can be loaded as a pytorch Dataset object from deepchecks.vision.datasets.segmentation, as is done in
```
Comment (noamzbr) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:40]: not Dataloader?

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 5]
```diff
@@ -27,151 +27,108 @@
...
+# Implementing the VisionData class
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 # The checks in the package validate the model & data by calculating various quantities over the data, labels and
 # predictions. In order to do that, those must be in a pre-defined format, according to the task type.
-# The first step is to implement a class that enables deepchecks to interact with your model and data and transform
-# them to this pre-defined format, which is set for each task type.
-# In this tutorial, we will implement the semantic segmentation task type by implementing a class that inherits from the
-# :class:`deepchecks.vision.segmentation_data.SegmentationData` class.
+# In the following example we're using pytorch. To see an implementation of this in tensorflow, please refer to
+# :ref:`creating VisionData guide <vision_data__creating_vision_data>`
```
Comment (noamzbr) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:67]: Again wrong link I think

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 6]
```diff
@@ -27,151 +27,108 @@
...
+# For pytorch, we will use our DataLoader, but we'll create a new collate function for it, that transforms the batch to
```
Comment (noamzbr) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:68]:
```suggestion
# For pytorch, we will use our DataLoader, but we'll create a new collate function for it that transforms the batch to
```

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 7]
```diff
@@ -27,151 +27,108 @@
...
-    def batch_to_images(self, batch):
-        """Convert the batch to a list of images, where each image is a 3D numpy array in the format (H, W, C)."""
-        return [tensor.numpy().transpose((1, 2, 0)) for tensor in batch[0]]
+    #predictions:
+    normalized_batch = [F.normalize(img.unsqueeze(0).float() / 255,
+                                    mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) for img in batch[0]]
+    predictions = [model(img)["out"].squeeze(0).detach() for img in normalized_batch]
+    predictions = [torch.nn.functional.softmax(pred, dim=0) for pred in predictions]
 
+    return {'images': images, 'labels': labels, 'predictions': predictions}
```
Comment (noamzbr) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:103]: Question - do we want to teach them to use `BatchOutputFormat` instead?

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 8]
```diff
@@ -197,3 +154,17 @@ def batch_to_images(self, batch):
 # Or, if working inside a notebook, the output can be displayed directly by simply printing the result object:
 
 result.show()
+
+# %%
+# From these results, we can see that mostly our model performs well. However, the model had an issue with identifying
```
Comment (noamzbr) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:159]: This is one big chunk of text... break it down into paragraphs? shorten it?

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 9]
```diff
@@ -27,151 +27,108 @@
...
+    images = [tensor.numpy().transpose((1, 2, 0)) for tensor in batch[0]]
 
-        return predictions
+    #labels:
+    labels = batch[1]
 
-    def batch_to_images(self, batch):
-        """Convert the batch to a list of images, where each image is a 3D numpy array in the format (H, W, C)."""
-        return [tensor.numpy().transpose((1, 2, 0)) for tensor in batch[0]]
+    #predictions:
```
Comment (Nadav-Barak) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:97]: maybe state what is the required format for predictions

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 10]
```diff
@@ -27,151 +27,108 @@
...
-from torchvision.utils import draw_segmentation_masks
+from deepchecks.vision.datasets.segmentation.segmentation_coco import CocoSegmentationDataset, load_model
 
-from deepchecks.vision.datasets.segmentation.segmentation_coco import load_dataset, load_model
-from deepchecks.vision.segmentation_data import SegmentationData
-
-train_loader = load_dataset(object_type='DataLoader', train=True)
-test_loader = load_dataset(object_type='DataLoader', train=False)
+train_dataset = CocoSegmentationDataset.load_or_download(train=True)
+test_dataset = CocoSegmentationDataset.load_or_download(train=False)
```
Comment (Nadav-Barak) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:49]: can add some prints (similar to other tutorials)

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py [hunk 11]
```diff
@@ -27,151 +27,108 @@
...
-        normalized_batch = [F.normalize(img.unsqueeze(0).float() / 255,
-                                        mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) for img in batch[0]]
-
-        predictions = [model(img)["out"].squeeze(0).detach() for img in normalized_batch]
-        predictions = [torch.nn.functional.softmax(pred, dim=0) for pred in predictions]
+    # images:
+    images = [tensor.numpy().transpose((1, 2, 0)) for tensor in batch[0]]
 
-        return predictions
+    #labels:
```
Comment (Nadav-Barak) [inline docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:94]: mention what is the label format (or mention it in the dataset creation part)

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py (hunk 4 above)
Comment (nirhutnik) [reply docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:40]: No, changed it, so I can create the dataloader with a specialized collate func

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py (hunk 9 above)
Comment (nirhutnik) [reply docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:97]: Discussed in person that no

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py (hunk 8 above)
Comment (nirhutnik) [reply docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:159]: Fixed

File: docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py (hunk 10 above)
Comment (nirhutnik) [reply docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py:49]: Done

**Review by noamzbr — APPROVED** [review]
